lib/screens/sub_category_list_view.dart

```diff
@@ -28,77 +33,72 @@ class _SubcategoryListState extends State<SubcategoryList> {
   void _initData() async {
     setState(() => _isLoading = true);
     String id = widget.categoryId;
-    String idname = widget.categoryName;
-    print(".....id matching or not.::$id......::$idname");
-    _subcategories = await CategoryRepo().getSubcategoriesByCategoryId(id);
-    _topics = await TopicRepo().getTopicsForCategoryId(_subcategories.first.id);
-
-    print(".......::database data is....${_topics.length}");
 
+    _subcategories = await CategoryRepo().getSubcategoriesByCategoryId(id);
```

Comment:
Do not prefetch all the topics in initState. When the user clicks on a tab, then fetch it